src/main/java/com/ocient/cli/CLI.java

```diff
@@ -1763,4 +1801,46 @@ private static void update(final String cmd)
 			lastCommandErrored = true;
 		}
 	}
+	private static void printToSession(String t) {
+		if(!(outputSessionFile.equals("") || (outputSessionFile.equals("stdout")))) {
+			try {
+				PrintStream standard = System.out;
+				File logFile = new File(outputSessionFile);
+				PrintStream logOut = new PrintStream(new FileOutputStream(logFile, true));
+				System.setOut(logOut);
+				System.setErr(logOut);
+				System.out.println("Ocient> " + t);
+
+				System.setOut(standard);
+				System.setErr(standard);
+			} catch(Exception e) {
+				System.out.println("Error: " + e.getMessage());
+			}
+		}
+	}
+
+	public static class TeeStream extends PrintStream {
+		PrintStream fileOutput;
+		PrintStream standardOutput;
+
+		public TeeStream(PrintStream out1, PrintStream out2) {
+			super(out1);
+			this.standardOutput = out1;
+			this.fileOutput = out2;
+		}
+		@Override
+		public void write(byte buf[], int off, int len) {
+			try {
+				standardOutput.write(buf, off, len);
+				fileOutput.write(buf, off, len);
```

Comment:
Assuming `fileOutput` is the passed to the super constructor, I still don't understand why this doesn't work
```suggestion
				super.write(buf, off, len);
```

Reply:
Just changed it, this does work! I'm not sure why this doesn't work with standardOutput (which is what I originally used)